src/town_gui.cpp

```diff
@@ -1293,6 +1287,25 @@ struct FoundTownWindow : Window {
 		}
 	}
 
+	void OnQueryTextFinished(std::optional<std::string> str) override
+	{
+		/* Was 'cancel' pressed? */
+		if (!str.has_value()) return;
+
+		int32_t value = 0;
+		if (!str->empty()) {
+			value = atoi(str->c_str());
+		}
```

Comment:
We are also trying to get rid of C string parsing functions. Whatever you write here, it would likely get changed soon again :)
Anyhow, `atoi` is the worst option, use `strtoul` for now?
```suggestion
		if (str->empty()) return;
		uint value = strtoul(str->c_str(), nullptr, 10);
```